download/download.go

```diff
@@ -37,17 +38,21 @@ var ArgHelp = `
 // Args is a flagset that needs to be exported so that it can be written to the
 // main program help
 var Args = flag.NewFlagSet("download", flag.ExitOnError)
+var outDir = Args.String("outdir", "", "Directory for downloaded files")
 
 // Gets the file name for a URL, using regex
-func createFilePathFromURL(file string) (fileName string, err error) {
+func createFilePathFromURL(file string, baseDir string) (fileName string, err error) {
 	// Create the file path according to the way files are stored in S3
 	// The folder structure comes after the UID described in the regex
 	re := regexp.MustCompile(`(?i)[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/(.*)`)
 	match := re.FindStringSubmatch(file)
 	if match == nil || len(match) < 1 {
 		return fileName, fmt.Errorf("failed to parse url for downloading file")
 	}
-	fileName = match[1]
+	if baseDir != "" && !strings.HasSuffix(baseDir, "/") {
+		baseDir += "/"
+	}
+	fileName = baseDir + match[1]
```

Comment:
It's not important, but might be neater to do this like:
```
    fileName  = filepath.Join(baseDir, match[1])
```
That's probably more platform safe as well.